staging/src/k8s.io/legacy-cloud-providers/azure/azure_controller_common.go

```diff
@@ -426,3 +435,14 @@ func getValidCreationData(subscriptionID, resourceGroup, sourceResourceID, sourc
 		SourceResourceID: &sourceResourceID,
 	}, nil
 }
+
+func isInstanceNotFoundError(err error) bool {
+	if strings.Contains(strings.ToLower(err.Error()), errVMSSInstanceNotFound) {
```

Comment:
Can we validate on the error code instead? or the "InvalidParameter" aspect of it. I'm not big fan of error message comparisons

Reply:
Are there any other cases when you'd get a `400` - if the resource you're trying to call is valid.

Reply:
can only do the error msg matching here? maybe add more matching condition?

Reply:
fix comments, PTAL

Reply:
I'd prefer if we check on `Code="InvalidParameter"` and `Target="instanceIds"` instead. My main concern is them changing the error message and breaking us in some way. 

Not gonna block on that however - current still works.

Reply:
fixed, PTAL